### ISSUE TITLE
`make check` fails on `typing_test_data.py`

### ISSUE THREAD

**Issue by jakobandersen** [sphinx-doc/sphinx#2020]
When I run `make check` I get the following error:

```
tests/typing_test_data.py:0: not compilable: invalid syntax (typing_test_data.py, line 6)
```

Should `typing_test_data.py` be excluded from `utils/check_sources.py`?

**Comment by shimizukawa** [comment]
It seems it works only with python3.
IMO, `typing_test_data.py` should be excluded on py2 environment and py3 environment without typing package.

@vlasovskikh how do you think?

**Comment by tk0miya** [comment]
I got same error through `tox` command with python2.7 in `master` branch (it seems `stable` branch is fine).
And I give `-- -d -m '^[tT]est'` options to `tox` command, it just resolved.

@shimizukawa I hope my case will help you.

**Comment by vlasovskikh** [comment]
@shimizukawa The `check_syntax` checker tries to compile its argument if it's Python 2. The `typing_test_data.py` file has to use Python 3 syntax, so it should be excluded somehow. For example, in `tests/tests_autodoc.py` in `test_type_hints` I use a conditional import and skip the test if the import fails. In the case of `make check` it's preferable to drop just this one check, but the only way I've found so far after a quick search is to add it to `DONT_CHECK` in `Makefile` and disable all checks.

**Comment by shimizukawa** [comment]
@tk0miya I've fixed it at 5b7ce8d.
@vlasovskikh Thanks, I disabled check for typing_test_data.py.
